### ISSUE TITLE
decrypt_dpapi_pw fails when knitting with knitr but not when running chunk

### ISSUE THREAD

**Issue by tphilippi** [yihui/knitr#1560]
In an Rmarkdown.Rmd document under RStudio 1.1.423 MS windows 7 or 10, calls to keyringr::decrypt_dpapi_pw() work when running individual or all chunks, but fail/hang (silently) when knitting the document in R-Studio under MS Windows.
My real use-case is passwords for database connections.  This is the minimal example .Rmd file, though it won't work on your computer unless you have keyringr set up and change the text in the call to decrypt_dpapi_pw()

I might bet a beer that there is something going on with file paths, and that the issue may be more with knitr or rmarkdown than with keyringr, but I'd be prepared to lose and gladly pay off the beer.  
I'm posting this issue here, also onhttps://github.com/cran/keyringr/issues and https://github.com/rstudio/rmarkdown/issues, and emailing Hadley as he works for/with RStudio and recommends keyringr in https://cran.r-project.org/web/packages/httr/vignettes/secrets.html

2 files pasted inline: test_keychain3.Rmd, then a copy/paste from the log file from running the chunks manually.
___
### test_keychain3.Rmd
---
title: "test_keyringr3"
author: "Tom 0"
date: "June 18, 2018"
output: html_document
---
Note that I slightly changed how I use credential_path from the vignette.  I have it point to the directory for the passwords with a trailing slash, so the paste0 below is the actual encrypted file.  
```{r setup}
library(knitr) # explicit to get version in sessionInfo()
library(rmarkdown) # explicit to get version
library(keyringr)
sessionInfo()
cat(credential_path, "\n")
cat(paste0(credential_path,"pgpwd.txt"),"\n")
pwd <- keyringr::decrypt_dpapi_pw(paste0(credential_path,"pgpwd.txt"))
print(pwd)
```

___
### log file
```
R version 3.5.0 (2018-04-23) -- "Joy in Playing"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(knitr) # explicit to get version in sessionInfo()
> library(rmarkdown) # explicit to get version
> library(keyringr)
> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] keyringr_0.4.0 rmarkdown_1.9  knitr_1.20    

loaded via a namespace (and not attached):
 [1] compiler_3.5.0  backports_1.1.2 magrittr_1.5    rprojroot_1.3-2 htmltools_0.3.6 tools_3.5.0    
 [7] yaml_2.1.18     Rcpp_0.12.16    stringi_1.1.7   stringr_1.3.0   digest_0.6.15   evaluate_0.10.1
[13] fortunes_1.5-5 
> cat(credential_path, "\n")
C:\Users\tphilippi\DPAPI\passwords\INP2300FCW18636\
> cat(paste0(credential_path,"pgpwd.txt"), "\n")
C:\Users\tphilippi\DPAPI\passwords\INP2300FCW18636\pgpwd.txt
> pwd <- keyringr::decrypt_dpapi_pw(paste0(credential_path,"pgpwd.txt"))
> print(pwd)
[1] "Plantago"
```

**Comment by yihui** [comment]
Duplicate of https://github.com/rstudio/rmarkdown/issues/1375

When you [cross-post issues](https://yihui.name/en/2017/10/cross-posting/), please mention it. Let's keep everyone in the same thread. Thanks!

**Comment by tphilippi** [comment]
Yihui--
I apologize.  I attempted to clearly list that I cross-posted in knitr, rmarkdown, and keyringr, as I don't have a clue which code base to even start looking in.  I did not get the issue numbers included in my links.  Next time I will post, then edit my postings to include issue numbers once they have them.

Thank you for the link to your post on better formatting for posting GitHub Issues that involve markdown.  I will use that in the future.
